pkg/authorization/authorizer/authorizer.go

```diff
@@ -28,13 +28,22 @@ type AuthorizationAttributeBuilder interface {
 }
 
 type AuthorizationAttributes interface {
+	// GetUserInfo returns the user requesting the action
 	GetUserInfo() user.Info
+	// GetVerb returns the verb associated with the action.  For resource requests, this verb is the logical kube verb.  For non-resource requests it is the http method tolowered.
 	GetVerb() string
+	// GetResource returns the resource type
 	GetResource() string
+	// GetNamespace returns the namespace of a resource request
 	GetNamespace() string
+	// GetResourceName returns the name of the resource being acted upon.  Not all resource actions have one (list as a for instance)
 	GetResourceName() string
 	// GetRequestAttributes is of type interface{} because different verbs and different Authorizer/AuthorizationAttributeBuilder pairs may have different contract requirements
 	GetRequestAttributes() interface{}
+	// IsNonResourceURL returns true if this is not an action performed against the resource API
+	IsNonResourceURL() bool
```

Comment:
also note implications for verb/namespace/resource/resourcename (verb is lowercased http verb, namespace/resource/resourcename are "")